### PR TITLE
Adds a c++17 build using the pre-release Ubuntu 18.04.

### PR DESCRIPTION
**Pull request by Pentarctagon**
Though given 00d87f8fe4078ffd84fcdc0d73fcc2104efef1e7 I would say this is better than nothing.

---

The checks for `Could not initialize SDL_video` were added since, for whatever reason, that error was [showing up at random](https://travis-ci.org/Pentarctagon/wesnoth/jobs/363709778#L2586) for everything *except* the mp tests.  The error never occurs twice in a row, [Xvfb](https://travis-ci.org/Pentarctagon/wesnoth/jobs/365380422#L2350) is running, and explicitly setting [SDL_VIDEODRIVER](https://wiki.libsdl.org/FAQUsingSDL#Linux) to x11 results in it instead saying that [x11 is not available](https://travis-ci.org/Pentarctagon/wesnoth/jobs/365380422#L3099).

Suffice it to say, I assume this is not a Wesnoth issue.

### CODE REVIEW

File: run_wml_tests [hunk 1]
```diff
@@ -188,6 +188,12 @@ run_test()
   fi
   $command 2> error.log
   error_code="$?"
+  # workaround to fix issue in travis+18.04
+  while [ "$(grep 'Could not initialize SDL_video' error.log)" != "" ]; do
```
Comment (jyrkive) [inline run_wml_tests]: Please use `grep -q` instead and check its return code:

```sh
while grep -q 'Could not initialize SDL_video' error.log do
```

File: utils/travis/play_test_executor.sh [hunk 2]
```diff
@@ -1,3 +1,12 @@
 #!/bin/bash
-set -e
-gdb -q -batch -return-child-result -ex "run" -ex "thread apply all bt" -ex "quit" --args ./wesnoth -m --controller 1:ai --controller 2:ai --nogui
+gdb -q -batch -return-child-result -ex "run" -ex "thread apply all bt" -ex "quit" --args ./wesnoth -m --controller 1:ai --controller 2:ai --nogui 2> error.log
+error_code="$?"
+while [ "$(grep 'Could not initialize SDL_video' error.log)" != "" ]; do
```
Comment (jyrkive) [inline utils/travis/play_test_executor.sh]: Same here.

File: utils/travis/test_executor.sh [hunk 3]
```diff
@@ -1,3 +1,12 @@
 #!/bin/bash
-set -e
-gdb -q -batch -return-child-result -ex "run" -ex "thread apply all bt" -ex "quit" --args ./boost_unit_tests
+gdb -q -batch -return-child-result -ex "run" -ex "thread apply all bt" -ex "quit" --args ./boost_unit_tests 2> error.log
+error_code="$?"
+while [ "$(grep 'Could not initialize SDL_video' error.log)" != "" ]; do
```
Comment (jyrkive) [inline utils/travis/test_executor.sh]: And here.

**Comment by Pentarctagon** [comment]
Alright, should be good now.

**Comment by Pentarctagon** [comment]
Not entirely sure why those errors are appearing for the C++17.  `src/serialization/base64.cpp` hasn't changed in about a month, but it was [working](https://travis-ci.org/Pentarctagon/wesnoth/jobs/363535834) on my fork's branch from just over a week ago(0e5463fd818b8cce388142c3ea949a289f6340ef).

**Comment by jyrkive** [comment]
I think it's missing `#include <string>`.

**Pull request merged**